cmd/web-handlers.go

```diff
@@ -791,9 +789,28 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	reader := r.Body
+	var reader io.Reader = r.Body
+	hashReader, err := hash.NewReader(reader, size, "", "", size)
+	if err != nil {
+		writeWebErrorResponse(w, err)
+		return
+	}
+	if objectAPI.IsEncryptionSupported() {
+		if hasServerSideEncryptionHeader(r.Header) && !hasSuffix(object, slashSeparator) { // handle SSE requests
+			reader, _, err = EncryptRequest(hashReader, r, bucket, object, metadata)
+			if err != nil {
+				writeWebErrorResponse(w, err)
+				return
+			}
+			info := ObjectInfo{Size: size}
+			hashReader, err = hash.NewReader(reader, info.EncryptedSize(), "", "", size) // do not try to verify encrypted content
+			if err != nil {
+				writeWebErrorResponse(w, err)
+				return
+			}
+		}
+	}
```

Comment:
@aead, please check this code - this is not getting encrypted though the metadata is updated correctly with encryption headers.